Bugfix/rusu1/visit 1901 expression added without applying

rusu24edward opened the pull request:
### Description

Resolves #1901

Adding and deleting expressions do not notify the gui until Apply is clicked.


### Type of change
Bugfix

### How Has This Been Tested?

No test in the test suite because this is a gui change. I tested it with several expressions and combinations of actions.

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have assigned reviewers (see [VisIt's PR procedures](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers) for more information).

rusu24edward committed:
Adding and deleting expressions will not notify the viewer but will still update the expression window

rusu24edward committed:
Adding expressions will not notify the viewer but will still update the expression window

rusu24edward committed:
Added docstrings

rusu24edward committed:
Merge branch 'bugfix/rusu1/visit-1901-expression-added-without-applying' of github.com:visit-dav/visit into bugfix/rusu1/visit-1901-expression-added-without-applying

brugger1 commented on src/gui/QvisExpressionsWindow.C:
```diff
@@ -1086,6 +1087,9 @@ QvisExpressionsWindow::apply()
 //
+//    Eddie Rusu, Wed Jun 24 15:46:57 PDT 2020
+//    Adding and deleting expressions will not notify the viewer and will still
+//    update the expressions window.
```
I'm not sure I understand your modification comment. I think it describes the bug you are fixing, not the solution.

rusu24edward replied:
Okay, how about now?

brugger1 replied:
Your new comment is much clearer.

brugger1 commented on src/gui/QvisExpressionsWindow.C:
```diff
@@ -1146,6 +1150,9 @@ QvisExpressionsWindow::addExpression()
 //
+//    Eddie Rusu, Wed Jun 24 15:46:57 PDT 2020
+//    Adding and deleting expressions will not notify the viewer and will still
+//    update the expressions window.
```
Same comment as earlier.

brugger1 replied:
Your new comment is much clearer.

brugger1 reviewed (changes requested):
You should probably also add something to the release notes.

rusu24edward committed:
Changed docstring to try to make more sense

rusu24edward committed:
Merge branch 'develop' of github.com:visit-dav/visit into bugfix/rusu1/visit-1901-expression-added-without-applying

rusu24edward commented:
> You should probably also add something to the release notes.

Ah, good catch. I put that on the PR for the RC branch #4859

brugger1 reviewed (changes requested):
You need to put the release notes changes on develop as well.

rusu24edward committed:
Updated relnotes

rusu24edward commented:
> You need to put the release notes changes on develop as well.

Ah, okay. I didn't know this about merging to develop. I added the release notes, and I also added the release notes from a merged PR from earlier this week where I made the same mistake.
